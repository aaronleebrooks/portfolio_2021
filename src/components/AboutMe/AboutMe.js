import styles from "./index.module.css";
import cx from "classnames";
import Link from "../Link";
import ResumePdf from "../../assets/pdf/resume.pdf";


const theme = {
  root: cx(
    styles.root
  ),
  textContainer: cx(
    styles.textContainer
  ),
  subText: cx(
    styles.text,
    styles.subText
  ),
  mainText: cx(
    styles.text,
    styles.mainText
  ),
  resumeBtn: cx(
    styles.resumeBtn
  ),
  photoContainer: cx(
    styles.photoContainer
  ),
  photo: cx(
    styles.photo
  ),
  downloadIcon: cx(
    styles.downloadIcon
  )
}

function AboutMe() {
  return (
    <div className={theme.root} id="about-me">
      <div className={theme.textContainer}>
        <span className={theme.subText}>Words, Words, Words</span>
        <span className={theme.mainText}>I do things that are good and cool.</span>
        <Link className={theme.resumeBtn} href={ResumePdf} download="resume.pdf" target='_blank'>
          My Resume <span className={theme.downloadIcon} />
        </Link>
      </div>
      <div className={theme.photoContainer}>
        <img className={theme.photo} alt="me" src="https://i.insider.com/5484d9d1eab8ea3017b17e29?width=600&format=jpeg&auto=webp" />
      </div>
    </div>
  );
}

export default AboutMe;
