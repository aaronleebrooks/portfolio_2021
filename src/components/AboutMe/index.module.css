.root {

    display: flex;
    font-family: Inter,sans-serif;
    padding: 40px;
}

.textContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    width: 65%;
}

.text {
    font-weight: 700;
    color: rgb(22, 28, 45);
}

.subText {
    font-size: 0.8125rem;
    line-height: 1.63;
    letter-spacing: 1.63px;
    margin-bottom: 5%;
    text-transform: uppercase;
}

.mainText {
    font-size: 76px;
    line-height: 84px;
    letter-spacing: -2.81px;
}

.resumeBtn {
    margin-top: 5%;
    cursor: pointer;
    font-size: 13px;
    font-weight: 700;
    letter-spacing: 1.63px;
    text-transform: uppercase;
    padding: 23px 30px;
    box-shadow: rgba(8, 7, 7, 0.12) 0px 32px 54px;
    border-radius: 50rem;
    display: inline-flex;
    -webkit-box-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    justify-content: space-between;
    user-select: none;
    transform: perspective(1px) translateZ(0px);
    position: relative;
    overflow: hidden;
    border: none;
    white-space: nowrap;
    color: rgb(255, 255, 255);
    background-color: rgb(73, 95, 239);
    transition: all 0.4s ease-out 0s;
    outline: none !important;
}

.photoContainer {
    display: flex;
    justify-content: center;
    align-items: center;
}

.photo {
    width: 25%;
}
