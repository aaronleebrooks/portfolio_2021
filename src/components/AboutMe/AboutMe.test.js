import { render } from '@testing-library/react';
import AboutMe from './AboutMe';

describe('AboutMe', () => {
  test('renders', () => {
    const { container } = render(<AboutMe />);
    expect(container).toMatchSnapshot();
  });
})
