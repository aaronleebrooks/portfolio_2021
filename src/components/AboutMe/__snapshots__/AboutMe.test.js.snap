// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe renders 1`] = `
<div>
  <div
    class="root"
    id="about-me"
  >
    <div
      class="textContainer"
    >
      <span
        class="text subText"
      >
        Words, Words, Words
      </span>
      <span
        class="text mainText"
      >
        I do things that are good and cool.
      </span>
      <a
        class="link resumeBtn"
        href="resume.pdf"
        rel="noreferrer"
        target="_blank"
      >
        My Resume 
        <span
          class="downloadIcon"
        />
      </a>
    </div>
    <div
      class="photoContainer"
    >
      <img
        alt="me"
        class="photo"
        src="https://i.insider.com/5484d9d1eab8ea3017b17e29?width=600&format=jpeg&auto=webp"
      />
    </div>
  </div>
</div>
`;
